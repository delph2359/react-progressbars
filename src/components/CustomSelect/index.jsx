import React, { Component } from "react";
import PropTypes from "prop-types";

export default class CustomSelect extends Component {
  static propTypes = {
    options: PropTypes.array,
    onChange: PropTypes.func
  };

  render() {
    const { options, onChange } = this.props;
    return (
      <select onChange={onChange} data-testid="custom-select">
        {options.map(option => (
          <option key={option.value} value={option.value}>
            {option.text}
          </option>
        ))}
      </select>
    );
  }
}
