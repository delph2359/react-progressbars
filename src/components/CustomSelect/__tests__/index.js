import "@testing-library/jest-dom/extend-expect";
import React from "react";
import { render } from "@testing-library/react";
import CustomSelect from "..";

describe("Custom Select", () => {
  test("should display options and values correctly", () => {
    const options = [
      { value: 1, text: "#progress1" },
      { value: 2, text: "#progress2" }
    ];
    const { container, getByText } = render(
      <CustomSelect options={options} onChange={() => console.log("changed")} />
    );

    expect(container).toMatchSnapshot();
    const progress1 = getByText("#progress1");
    expect(progress1).toBeInTheDocument();
    expect(progress1.value).toEqual("1");
    const progress2 = getByText("#progress2");
    expect(progress2).toBeInTheDocument();
    expect(progress2.value).toEqual("2");
  });
});
