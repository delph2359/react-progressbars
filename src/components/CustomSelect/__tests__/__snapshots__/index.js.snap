// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Select should display options and values correctly 1`] = `
<div>
  <select
    data-testid="custom-select"
  >
    <option
      value="1"
    >
      #progress1
    </option>
    <option
      value="2"
    >
      #progress2
    </option>
  </select>
</div>
`;
